frontend/include/chpl/resolution/scope-types.h

```diff
@@ -478,12 +471,26 @@ class VisibilitySymbols {
     CONTENTS_EXCEPT,
   };
 
+  /** Indicating shadow scopes */
+  enum ShadowScope {
+    // public use / private import / public import all work at the same
+    // scope level as symbols declared within a module, and
+    // REGULAR_SCOPE indicates that.
+    REGULAR_SCOPE = 0,
+
+    // 'private use' aka just 'use' introduces 2 shadow scopes
+    // that appear to be between the module and any parent symbols
+    SHADOW_SCOPE_ONE = 1, // for module contents brought in by a 'private use'
```

Comment:
Would more descriptive names be better here, like `CONTENTS_SHADOW_SCOPE` and `NAME_SHADOW_SCOPE`? It's not like the scopes are numbered in the spec.